build/maven.go

```diff
@@ -172,42 +182,93 @@ func (mm *MavenModule) loadMavenHome() (mavenHome string, err error) {
 		// Since Maven installation can be located in different locations,
-		// Depending on the installation type and the OS (for example: For Mac with brew install: /usr/local/Cellar/maven/{version}/libexec or Ubuntu with debian: /usr/share/maven),
-		// We need to grab the location using the mvn --version command
-
+		// depending on the installation type and the OS (for example: For Mac with brew install: /usr/local/Cellar/maven/{version}/libexec or Ubuntu with debian: /usr/share/maven),
+		// we need to grab the location using the mvn --version command.
 		// First we will try lo look for 'mvn' in PATH.
-		mvnPath, err := exec.LookPath("mvn")
-		if err != nil || mvnPath == "" {
-			return "", errors.New(err.Error() + "Hint: The mvn command may not be included in the PATH. Either add it to the path, or set the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
+		maven, err := mm.getExecutableName()
+		if err != nil {
+			return maven, err
 		}
-		mm.containingBuild.logger.Debug(MavenHome, " is not defined. Retrieving Maven home using 'mvn --version' command.")
-		cmd := exec.Command("mvn", "--version")
-		var stdout bytes.Buffer
-		cmd.Stdout = &stdout
-		err = cmd.Run()
+		if !mm.extractorDetails.useWrapper {
+			mvnPath, err := mm.lookPath()
+			if err != nil {
+				return mvnPath, err
+			}
+		}
+		versionOutput, err := mm.execMavenVersion(maven)
 		if err != nil {
 			return "", err
 		}
-		output := strings.Split(strings.TrimSpace(stdout.String()), "\n")
 		// Finding the relevant "Maven home" line in command response.
-		for _, line := range output {
-			if strings.HasPrefix(line, "Maven home:") {
-				mavenHome = strings.Split(line, " ")[2]
-				if runtime.GOOS == "windows" {
-					mavenHome = strings.TrimSuffix(mavenHome, "\r")
-				}
-				mavenHome, err = filepath.Abs(mavenHome)
-				if err != nil {
-					return "", err
-				}
-				break
-			}
-		}
-		if mavenHome == "" {
-			return "", errors.New("Could not find the location of the maven home directory, by running 'mvn --version' command. The command output is:\n" + stdout.String() + "\nYou also have the option of setting the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
+		mavenHome, err = mm.extractMavenPath(versionOutput)
+		if err != nil {
+			return "", err
 		}
 	}
 	mm.containingBuild.logger.Debug("Maven home location: ", mavenHome)
+
+	return
+}
+
+func (mm *MavenModule) lookPath() (mvnPath string, err error) {
+	mvnPath, err = exec.LookPath("mvn")
+	err = mm.determineError(mvnPath, "", err)
+	return
+}
+
+// For every different we show to users a message that helps him to solve the error
+// and this function is to determine which is the suitable message to show depend on the error we got
+func (mm *MavenModule) determineError(mvnPath string, versionOutput string, err error) error {
```

Comment:
determineError(mvnPath string, versionOutput string, err error) 
-->
determineError(mvnPath, versionOutput string, err error)

We should add CheckError to the errors returned by this function.